src/UglyToad.PdfPig/Writer/PdfStreamWriter.cs

```diff
@@ -151,21 +181,6 @@ private IndirectReferenceToken AddToken(IToken token, int reservedNumber)
             return referenceToken;
         }
 
-        private IndirectReferenceToken FindToken(IToken token)
```

Comment:
I removed this method, because It was causing a huge performance drop with big file. I need to come up with a better idea.

Reply:
I wrote something similar for a different library. During writing of tokens it would create a hash (FNV but not really important) of the written contents (excluding the object header and footer) and store it in a dictionary. Subsequent writes would check to see if the hash had been written and use the existing ref if it had. The hashing was so quick that it didn't affect performance noticeably for large documents.